src/Rings/MPolyMap/AffineAlgebras.jl

```diff
@@ -222,7 +222,43 @@ function preimage(F::MPolyAnyMap, I::Ideal)
 end
 
 function preimage(
-    f::MPolyAnyMap{<:MPolyRing{T}, CT}, 
+    f::MPolyAnyMap{<:MPolyRing{T}, CT, Nothing}, 
+    I::Union{MPolyIdeal, MPolyQuoIdeal}
+  ) where {T <: RingElem,
+           CT <: Union{MPolyRing{T}, MPolyQuoRing{<:MPolyRingElem{T}}}}
+  return _preimage_via_singular(f, I)
+end
+
+function preimage(
+    f::MPolyAnyMap{<:MPolyRing{T}, <:MPolyRing{T}, Nothing}, 
+    I::MPolyIdeal{T}
+  ) where {T <: RingElem}
+  # If the map is the inclusion of a subring in a subset of 
+  # variables, use the `eliminate` command instead.
+  if all(_is_gen, f.img_gens) && allunique(f.img_gens)
```

Comment:
Don't we already store this somewhere in `f.variable_indices`?

Reply:
Yes, looks like it. I already forgot about that...